File: bench/python/bintree.py
import os, sys
sys.setrecursionlimit(100000)
N = int(os.environ.get("BENCH_N", "40"))   # repetitions
DEPTH = 12

def make(d):
    if d == 0:
        return None
    return (make(d - 1), make(d - 1))

def check(node):
    if node is None:
        return 1
    return 1 + check(node[0]) + check(node[1])

total = 0
for _ in range(N):
    total += check(make(DEPTH))

print(total)

File: bench/node/bintree.js
const N = parseInt(process.env.BENCH_N || "40", 10);   // repetitions
const DEPTH = 12;

function make(d) {
  if (d === 0) return null;
  return [make(d - 1), make(d - 1)];
}

function check(node) {
  if (node === null) return 1;
  return 1 + check(node[0]) + check(node[1]);
}

let total = 0;
for (let i = 0; i < N; i++) total += check(make(DEPTH));

console.log(total);
